services/galley/src/Galley/API/Teams/Features.hs

```diff
@@ -200,16 +200,18 @@ pushFeatureConfigEvent ::
   Sem r ()
 pushFeatureConfigEvent tid event = do
   memList <- getTeamMembersForFanout tid
-  when ((memList ^. teamMemberListType) == ListTruncated) $ do
-    P.warn $
-      Log.field "action" (Log.val "Features.pushFeatureConfigEvent")
-        . Log.field "feature" (Log.val (toByteString' . Event._eventFeatureName $ event))
-        . Log.field "team" (Log.val (cs . show $ tid))
-        . Log.msg @Text "Fanout limit exceeded. Some events will not be sent."
-  let recipients = membersToRecipients Nothing (memList ^. teamMembers)
-  for_
-    (newPush (memList ^. teamMemberListType) Nothing (FeatureConfigEvent event) recipients)
-    push1
+  if ((memList ^. teamMemberListType) == ListTruncated)
+    then do
+      P.warn $
+        Log.field "action" (Log.val "Features.pushFeatureConfigEvent")
+          . Log.field "feature" (Log.val (toByteString' . Event._eventFeatureName $ event))
+          . Log.field "team" (Log.val (cs . show $ tid))
+          . Log.msg @Text "Fanout limit exceeded. Events will not be sent."
+    else do
+      let recipients = membersToRecipients Nothing (memList ^. teamMembers)
+      push $
+        maybeToList $
+          (newPush Nothing (toJSONObject event) recipients)
```

Comment:
These two don't have the same behavior. In the old ones we'd always have a push. Why the change?

Reply:
In the old code we were sending and argument of type `ListType` to `push`. In the logic for `push` if it saw it was a `ListTruncated` it used to just not push anything. I found this really odd and just removed this `ListType` and made this place make a decision about whether or not things should be pushed.
It turned out that this was the only place from where we could send a `ListTrucated`. 

I think this is a better approach because it is not a job of the notification subsystem to say that if a notification cannot be sent to all the people, noone gets it. It should be decided case by case.